contributing/code/stack_trace.rst

```diff
@@ -0,0 +1,182 @@
+Getting a Stack Trace
+=====================
+
+When :doc:`reporting a bug </contributing/code/bugs>` for an
+exception, it is instrumental that you provide one or several stack
+traces. To understand why, you first have to understand what a stack
+trace is, and how it can be useful to you as a developer.
+
+Anatomy of a Stack Trace
+------------------------
+
+A stack trace is called that way because it allows one to see a trail of
+function calls leading to a point in code since the beginning of the
+program. That point is not necessarily an exception. For instance, you
+can use the native PHP function ``debug_print_backtrace()`` to get such
+a trace. For each line in the trace, you get a file and a function or
+method call, and the line number for that call. This is often of great
+help for understanding the flow of your program and how it can end up in
+unexpected places, such as lines of code where exceptions are thrown.
+
+Stack Traces and Exceptions
+---------------------------
+
+In PHP, every exception comes with it own stack trace, which is
+displayed by default if the exception is not caught. When using Symfony,
+such exceptions go through a custom exception handler, which enhances
+them in various ways before displaying them according to the current
+Server API (CLI or not).
+This means a better way to get a stack trace when you do need the
+program to continue is to throw an exception, as follows:
+``throw new \Exception();``
+
+Nested Exceptions
+-----------------
+
+When programs get bigger, complexity is often tackled with layers of
+architecture that need to be kept separate. For instance, if you have an
+web application that makes a call to a remote API, it might be good to
+wrap exceptions thrown when making that call with exceptions that have
+special meaning in your domain, and to build appropriate HTTP exceptions
```

Comment:
```suggestion
special meaning in your domain, and to return appropriate HTTP exceptions
```

?

Reply:
Exceptions are not returned usually, they are thrown, and before that, built, which here I think is more appropriate since you build an exception from the previous exception.

Reply:
i see your point now